x-pack/plugins/rule_registry/server/plugin.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PluginInitializerContext, Plugin, CoreSetup } from 'src/core/server';
+import { PluginSetupContract as AlertingPluginSetupContract } from '../../alerting/server';
+import { RuleRegistry } from './rule_registry';
+import { defaultIlmPolicy } from './rule_registry/defaults/ilm_policy';
+import { defaultFieldMap } from './rule_registry/defaults/field_map';
+
+export type RuleRegistryPluginSetupContract = RuleRegistry<typeof defaultFieldMap>;
+
+export class RuleRegistryPlugin implements Plugin<RuleRegistryPluginSetupContract> {
+  constructor(private readonly initContext: PluginInitializerContext) {
+    this.initContext = initContext;
+  }
+
+  public setup(
+    core: CoreSetup,
+    plugins: { alerting: AlertingPluginSetupContract }
+  ): RuleRegistryPluginSetupContract {
+    const globalConfig = this.initContext.config.legacy.get();
```

Comment:
Is there a non legacy config we can use?

Reply:
I can look into that. Stole this from somewhere else, I think the event log.

Reply:
Looks like [most all usages](https://github.com/search?q=kibana.index+repo%3Aelastic%2Fkibana+filename%3Aplugin.ts&type=Code&ref=advsearch&l=&l=) throughout plugin initialization use `initContext.config.legacy.get()`, even [kibana security](https://github.com/elastic/kibana/blob/6ecffcc57c9a5b0328257cd25b737140a8471dda/x-pack/plugins/security/server/plugin.ts#L209). @kobelb are you aware of a non-legacy way or retrieving, or any other implications here with regards to https://github.com/elastic/cloud/issues/77641?